momento-sdk/src/main/java/momento/sdk/responses/storage/data/GetResponse.java

```diff
@@ -0,0 +1,86 @@
+package momento.sdk.responses.storage.data;
+
+import momento.sdk.exceptions.ClientSdkException;
+import momento.sdk.exceptions.SdkException;
+
+/** Response for a get operation */
+public interface GetResponse {
+
+  /** A successful get operation. */
+  class Success implements GetResponse {
+    private final Object value;
+    private final ValueType valueType;
+
+    private Success(Object value, ValueType valueType) {
+      this.value = value;
+      this.valueType = valueType;
+    }
+
+    public static Success of(byte[] value) {
+      return new Success(value, ValueType.BYTE_ARRAY);
+    }
+
+    public static Success of(String value) {
+      return new Success(value, ValueType.STRING);
+    }
+
+    public static Success of(long value) {
+      return new Success(value, ValueType.LONG);
+    }
+
+    public static Success of(double value) {
+      return new Success(value, ValueType.DOUBLE);
+    }
+
+    public ValueType getType() {
+      return valueType;
+    }
+
+    public byte[] getValueAsByteArray() {
+      ensureCorrectTypeOrThrowException(ValueType.BYTE_ARRAY, valueType);
+      return (byte[]) value;
+    }
+
+    public String getValueAsString() {
+      ensureCorrectTypeOrThrowException(ValueType.STRING, valueType);
+      return (String) value;
+    }
+
+    public long getValueAsLong() {
+      ensureCorrectTypeOrThrowException(ValueType.LONG, valueType);
+      return (long) value;
+    }
+
+    public double getValueAsDouble() {
```

Comment:
For consistency with the cache client, I could also rename these `valueAsByteArray`, `valueAsString`, ... etc though I feel `get ... As ...` is easier to read/more idiomatic to Java.

Reply:
Thoughts @cprice404 ?

Reply:
i think i probably agree with your assessment about which would be more idiomatic, but i think consistency has to win out here. thanks for checking.

Reply:
Renamed in [451f649](https://github.com/momentohq/client-sdk-java/pull/358/commits/451f6492c9f755fade41da4d26dc37eda9928352) to be idiomatic with cache response accessors.